manifests/sentinel.pp

```diff
@@ -89,78 +63,115 @@
+  Stdlib::Port $sentinel_port                 = 26379,
+  String[1] $service_group                    = 'redis',
+  String[1] $service_name                     = $redis::params::sentinel_service_name,
+  Stdlib::Ensure::Service $service_ensure     = 'running',
+  Boolean $service_enable                     = true,
+  String[1] $service_user                     = 'redis',
+  Stdlib::Absolutepath $working_dir           = $redis::params::sentinel_working_dir,
   Variant[Undef, Stdlib::IP::Address, Array[Stdlib::IP::Address]] $sentinel_bind = undef,
-  Stdlib::Port $sentinel_port = 26379,
-  String[1] $service_group = 'redis',
-  String[1] $service_name = $redis::params::sentinel_service_name,
-  Stdlib::Ensure::Service $service_ensure = 'running',
-  Boolean $service_enable = true,
-  String[1] $service_user = 'redis',
-  Stdlib::Absolutepath $working_dir = $redis::params::sentinel_working_dir,
-  Optional[Stdlib::Absolutepath] $notification_script = undef,
-  Optional[Stdlib::Absolutepath] $client_reconfig_script = undef,
 ) inherits redis::params {
 
   require 'redis'
 
   if $facts['os']['family'] == 'Debian' {
     package { $package_name:
       ensure => $package_ensure,
-      before => File[$config_file_orig],
+      before => Concat[$config_file_orig],
     }
 
     if $init_script {
       Package[$package_name] -> File[$init_script]
     }
   }
 
-  file { $config_file_orig:
-    ensure  => file,
-    owner   => $service_user,
-    group   => $service_group,
-    mode    => $config_file_mode,
-    content => template($conf_template),
+  Concat {
+    owner => $service_user,
+    group => $service_group,
+    mode  => $config_file_mode,
+  }
+
+  concat { $config_file_orig:
+    ensure => present,
+  }
+
+  concat::fragment { 'sentinel_conf_header':
+    target  => $config_file_orig,
+    order   => 10,
+    content => template('redis/sentinel/redis-sentinel.conf_header.erb'),
+  }
+
+  $sentinel_monitor.each |$monitor,$values| {
+    $_monitor = merge($monitor_defaults,$values)
+    $redis_values = merge({'monitor_name' => $monitor},$_monitor)
+    concat::fragment { "sentinel_conf_monitor_${monitor}" :
+      target  => $config_file_orig,
+      order   => 20,
+      content => epp('redis/sentinel/redis-sentinel.conf_monitor.epp', {
+        monitor_name           => $redis_values['monitor_name'],
+        redis_host             => $redis_values['redis_host'],
+        redis_port             => $redis_values['redis_port'],
+        quorum                 => $redis_values['quorum'],
+        down_after             => $redis_values['down_after'],
+        parallel_sync          => $redis_values['parallel_sync'],
+        failover_timeout       => $redis_values['failover_timeout'],
+        auth_pass              => $redis_values['auth_pass'],
+        notification_script    => $redis_values['notification_script'],
+        client_reconfig_script => $redis_values['client_reconfig_script'],
+      }),
```

Comment:
Just thinking out loud, can you pass it directly?
```suggestion
      content => epp('redis/sentinel/redis-sentinel.conf_monitor.epp', $redis_values)
```

Reply:
suggestion accepted :)